core/bessctl.cc

```diff
@@ -1017,10 +1017,28 @@ class BESSControlImpl final : public BESSControl::Service {
     response->mutable_inc()->set_dropped(stats.inc.dropped);
     response->mutable_inc()->set_bytes(stats.inc.bytes);
 
+    response->mutable_inc()->mutable_requested_hist()->Reserve(HISTOGRAM_SIZE);
+    response->mutable_inc()->mutable_requested_hist()->ExtractSubrange(0, HISTOGRAM_SIZE, stats.inc.requested_hist);
```

Comment:
Ugh... my bad. It seems that `ExtractSubrange()` copies from pb to array, not the other way around. From the documentation, the below code should work (not tested) I guess?
```
// no Reserve() is required
*response->mutable_inc()->mutable_requested_hist() = {stats.inc.requested_hist, stats.inc.requested_hist + HISTOGRAM_SIZE};
```

By the way, if you want to eliminate an implicit assumption -- the size of the array is HISTOGRAM_SIZE -- you can:
```
struct QueueStats {
    ...
    std::array<uint64_t, bess::PacketBatch::kMaxBurst + 1> requested_hist;
    ...
};
*response->mutable_inc()->mutable_requested_hist() = {stats.inc.requested_hist.begin(), stats.inc.requested_hist.end()};
```
It also removes the macro HISTOGRAM_SIZE. Having macros in the source code is a legacy from the C implementation in the past. Newly added C++ code should avoid using macros.